### ISSUE TITLE
Prevent segment + google analytics from discarding events due to missing `name` in the segment data field

### ISSUE THREAD

**Issue by zspencer** [expo/expo#1485]
In order to diagnose and tweak analytics instrumentation with close feedback loops
As a developer
I want to be able to see the events sent to google analytics via segment even when I'm running my app inside of the Expo  App.

### Environment

Environment:
  OS: macOS High Sierra 10.13.3
  Node: 9.3.0
  Yarn: 1.3.2
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  expo: ^25.0.0 => 25.0.0
  react: 16.2.0 => 16.2.0
  react-native: https://github.com/expo/react-native/archive/sdk-25.0.0.tar.gz => 0.52.0

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/zspencer-cdfcc687-0c87-4c28-ba6b-f06f0d234bd6.tar.gz

Currently, we are doing "daily" builds inside of the expo app, however we are targeting standalone.

### Steps to Reproduce
1. Set up an Expo to stream events to Segment and then Segment to stream events  to Google Analytics. (Note: You will want to use a "Web" property in google analytics)
2. Instrument an Identify or "Track" event
3. Notice that the event gets sent to Segment by inspecting the debugger, but the event does not get appear in Google Analytics (Use the "Live" view in google analytics so you can see events as they stream in)

This is because Google Analytics discards events that don't have an app name. The Expo client (from what I can tell) does not send the app name to Segment when running a non-standalone app.

This results in a significantly slowed-down feedback loop when building out analytics instrumentation targeting google analytics, as you have to publish to a standalone app on every change. It can also result in lost-time due to tracking down exactly why events aren't streaming through. (This issues existence should help with that though :))

See https://forums.expo.io/t/segment-integration-does-not-appear-to-pass-screen-events-to-google-analytics-web-properties/7537/3 for more information.

### Expected Behavior
I expected events to be visible in Google Analytics live event stream

### Actual Behavior
Events were not sent through to Google Analytics

### Reproducible Demo
A reproducible demo requires a Segment account configured to send events to a Google Analytics web property. The following app will do so if you provide appropriate Segment write keys: https://snack.expo.io/Sy38mJU_f

**Comment by fxfactorial** [comment]
hi there, possibly related to: https://github.com/expo/expo/issues/680?

**Comment by zspencer** [comment]
@fxfactorial yes and no, that resolves it for standalone apps, but does not resolve it for apps running inside of Expo. The Expo iOS app itself needs to set a display name when not running in standalone mode.

**Comment by fxfactorial** [comment]
@zspencer Cool thanks, question: is this because you don't want to go through test flight? Talked with team and I think we can add this eventually.

**Comment by zspencer** [comment]
It's more about being able to get feedback quickly when instrumenting analytics. As it stands, I have to do a publish to the build that's running testflight before I can see events going through to google. This increases the time between building the feature and verifying it works pretty significantly (I tend to do testflight builds once a day at most.)

It also can result in wild goose chases for people who are building and testing the app primarily through the Expo app; as events disappear without a trace. You wind up needing to have a lot more understanding of the internal protocol between Segment and Google Analytics than is useful.

All that said, this is much more of a "It'd be nice and it would probably save time for future people who are using the Expo iOS app to check that they're wiring Google Analytics/Segment in correctly" than it a breaking bug because standalone apps *do* send events to google analytics when an app.json name is configured.

TL/DR: Don't worry too much about it, I think this issue existing and being open will probably solve most of the initial time-loss and I may submit a patch for it this evening or weekend if I can get the Expo iOS app building locally later :)

**Comment by fxfactorial** [comment]
Appreciate your words and insights. I think anything that cuts the iteration time on test, publishing and getting good apps in peoples hands is important to the expo experience.

**Comment by terribleben** [comment]
Latest Expo Client should fix this 👍
